### PR TITLE
Add support for lerna

### PR DESCRIPTION
**Pull request by DrSensor**
Because the command [`lerna version`](https://github.com/lerna/lerna/tree/master/commands/version) doesn't bump the `version` field of root _package.json_, it only bump the `version` field of _lerna.json_ and _packages/*/package.json_

This update prioritizes reading the `version` field in _lerna.json_ if the file exists, otherwise will fallback for workPath/_package.json_

### CODE REVIEW

**Comment by jgravois** [comment]
these changes makes sense to me.

FWIW, we sidestepped the whole can of worms in our latest lerna project by just incrementing the package.json file version manually.

https://github.com/Esri/arcgis-rest-js/blob/8be88a399a35e1dc31f77f377fb095489e055b2f/support/publish.sh#L19-20

**Comment by bcomnes** [comment]
Up to @ungoldman

File: bin/lib/get-defaults.js [hunk 1]
```diff
@@ -43,13 +45,20 @@ function getDefaults (workPath, isEnterprise, callback) {
...
+        return callback(new Error(errStr))
+      }
+    } else if (log.version !== pkg.version) {
+      errStr = 'CHANGELOG.md out of sync with package.json '
       errStr += '(' + (log.version || log.title) + ' !== ' + pkg.version + ')'
       return callback(new Error(errStr))
     }
 
-    var version = pkg.version ? 'v' + pkg.version : null
+    var version = pkg.version ? 'v' + pkg.version : lerna.version ? 'v' + lerna.version : null
```
Comment (ungoldman) [inline bin/lib/get-defaults.js]: if for some reason `pkg.version` is falsey and `lernaPath` doesn't exist, `lerna.version` will throw an uncaught error. can you construct this in a way that doesn't risk accessing a property on an undefined variable?

Comment (DrSensor) [reply bin/lib/get-defaults.js]: updated

Comment (ungoldman) [reply bin/lib/get-defaults.js]: thank you!

**Comment by ungoldman** [comment]
🚀 [`gh-release@3.5.0`](https://github.com/hypermodules/gh-release/releases/tag/v3.5.0)

**Pull request merged**